src/StateService/Verification/VerificationService.cs

```diff
@@ -0,0 +1,161 @@
+using Akka.Actor;
+using Akka.Util.Internal;
+using Neo.IO;
+using Neo.Network.RPC;
+using Neo.Plugins.StateService.Storage;
+using Neo.Wallets;
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Neo.Plugins.StateService.Verification
+{
+    public class VerificationService : UntypedActor
+    {
+        public class ValidatedRootPersisted { public uint Index; }
+        public class BlockPersisted { public uint Index; }
+        private class Timer { public uint Index; }
+        private static readonly uint TimeoutMilliseconds = ProtocolSettings.Default.MillisecondsPerBlock;
+        private static readonly uint DelayMilliseconds = 3000;
+        private readonly NeoSystem core;
+        private const int MaxCachedVerificationProcessCount = 10;
+        private readonly Wallet wallet;
+        private readonly ConcurrentDictionary<uint, VerificationContext> contexts = new ConcurrentDictionary<uint, VerificationContext>();
+
+        public VerificationService(NeoSystem core, Wallet wallet)
+        {
+            this.core = core;
+            this.wallet = wallet;
+        }
+
+        private void SendVote(VerificationContext context)
+        {
+            var vote = context.CreateVote();
+            if (vote is null) return;
+            Utility.Log(nameof(VerificationService), LogLevel.Info, $"relay vote, height={vote.RootIndex}, retry={context.Retries}");
+            Settings.Default.VerifierUrls.ForEach(url =>
+            {
+                Task.Run(() =>
+                {
+                    try
+                    {
+                        var http_client = new HttpClient
+                        {
+                            Timeout = TimeSpan.FromMilliseconds(TimeoutMilliseconds),
+                        };
+                        var client = new RpcClient(http_client, url);
+                        client.RpcSendAsync("votestateroot", vote.RootIndex, vote.ValidatorIndex, Convert.ToBase64String(vote.Signature))
```

Comment:
Who are able to send Votes? Which nodes can connect to the Verifies and send valid votes?

Reply:
I also did not fully understand who will determine the Verifiers.

Reply:
> Initially its empty right until we vote and fill NativeContract.RoleManagement.GetDesignatedByRole.?

Yes


> I also did not fully understand who will determine the Verifiers.


The state verifier is designated by the committee.

Reply:
The Default values on the  `config.json` are just used to start a communication between them, right?

Reply:
In this sense, we would consider that the `VerifierUrls` is fixed to `4` and `Committee` would keep this number as `4` to be consistent with the Default `Config`.

Finally, thus, a normal user will just start the service with `start states` and it will sync normally and then the user will have a personalized RPC, right?

Reply:
> Who are able to send Votes? Which nodes can connect to the Verifies and send valid votes?

Normally only verifiers send votes, but anyone can. Because it's a rpc service, anyone can connect and send once the verifiers' rpc port exposed. But only verifiers vote is valid.

> Initially its empty right until we vote and fill `NativeContract.RoleManagement.GetDesignatedByRole`.?

The `VerifierUrls` is not saved in `RoleManagement` .Committee only designate verifier public keys. Verifier urls should be manully changed in config.json of this service.

> Finally, thus, a normal user will just start the service with `start states` and it will sync normally and then the user will have a personalized RPC, right?

Normal user don't need to start states when sync. They can state states but there will be nothing because it's not verifier which are designated in `RoleManagement`.

Reply:
@ZhangTao1596, thus, how the system can scale up?

I mean, a project running on Neo that wants to follow the `states` needs to start and sync, right?
Then it would have its personalized RPC for querying data from.
Otherwise, the officials RPC workload would be too much.

Reply:
> I mean, a project running on Neo that wants to follow the `states` needs to start and sync, right?
> Then it would have its personalized RPC for querying data from.
> Otherwise, the officials RPC workload would be too much.

Right. Normal node installed state service will have state releated rpc service. They don't need to query verifier or official rpc nodes.
Futhermore normal nodes don't need to `start states`. Once the plugin installed, sync and rpc service will work. Considering mpt calculating, node should install this plugin before block syncing.
Verifiers use `start states` to start verifying.

Reply:
@ZhangTao1596, sounds good.

In this sense, @ZhangTao1596.
Maybe there should be  warning if the plugin is installed in a node that already has synced some blocks.
Perhaps this could be done in another PR.

Thanks you and @Tommo-L for the clarifications.

Great work!